src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,191 @@
+
+    const tiff = await loadFirstTiff(args.source, collection);
+
+    const category = getCategory(collection);
+    const geographicDescription = getGeographicDescription(collection);
+    const region = getRegion(collection);
+    const event = getEvent(collection);
+    const date = getDate(collection);
+    const gsd = extractGsd(tiff);
+    const epsg = extractEpsg(tiff);
+
+    const target = generatePath(args.targetBucketName, category, geographicDescription, region, event, date, gsd, epsg);
+    logger.info({ duration: performance.now() - startTime, target: target }, 'GeneratePath:Done');
+    return target;
+  },
+});
+
+export function generatePath(
+  targetBucketName: string,
+  category: string,
+  geospaital_description: string,
+  region: string,
+  event: string,
+  date: string,
+  gsd: string,
+  epsg: string,
+): string {
+  const name = generateName(region, geospaital_description, event);
+
+  if (category === dataCategories.SCANNED_AERIAL_PHOTOS) {
+    // nb: Historic Imagery is out of scope as survey number is not yet recorded in collection metadata (15/02/24)
+    throw new Error(`Automated target generation not implemented for historic imagery`);
+  } else if ([dataCategories.URBAN_AERIAL_PHOTOS, dataCategories.RURAL_AERIAL_PHOTOS].includes(category)) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
+  } else if (category === dataCategories.SATELLITE_IMAGERY) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
+  } else if ([dataCategories.DEM, dataCategories.DSM].includes(category)) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}/${category}_${gsd}/${epsg}/`;
+  } else {
+    throw new Error(`Path Can't be generated from collection as no matching category: ${category}.`);
+  }
+}
+
+export function generateName(region: string, geospaital_description: string, event: string): string {
+  /* Function to:
+    - Determine if region or geospatial description should be the dataset name
+    - Add event to name
+  */
+  let name = region;
+  if (geospaital_description) {
+    name = geospaital_description.toLowerCase().replace(/\s+/g, '-');
+  }
+  if (event) {
+    name = `${name}-${event.toLowerCase().replace(/\s+/g, '-')}`;
+  }
+  return name;
+}
+
+export function getCategory(collection: StacCollection): string {
+  const category = collection['linz:geospatial_category'] as string;
+  if (!category) {
+    throw new Error('No category in collection');
+  }
+  return category;
+}
+
+export function getGeographicDescription(collection: StacCollection): string {
+  // This is optional metadata, therefore returning nothing is ok.
+  return collection['linz:geographic_description'] as string;
+}
+
+export function getEvent(collection: StacCollection): string {
+  // This is optional metadata, therefore returning nothing is ok.
+  return collection['linz:event_name'] as string;
```

Comment:
What happens if `linz:event_name` is not set? Wouldn't that be an error?

Also, what if `linz:event_name` is set to `null`? Does it get coerced to an empty string or to `"null"`?

Reply:
```
const collection = await fsa.readJson<StacCollection>('./src/commands/path/__test__/sample.json');
collection['linz:event_name'] = null;
collection['linz:geographic_description] = null;
const event = getEvent(collection);
const gd = getGeographicDescription(collection);
const target = generatePath('nz-imagery', 'urban-aerial-photos', gd, 'hawkes-bay', event, '2017-2018', '0.05m', '2193');
console.log(target);
// s3://nz-imagery/hawkes-bay/hawkes-bay_2017-2018_0.05m/rgb/2193/
```

```
const collection = await fsa.readJson<StacCollection>('./src/commands/path/__test__/sample.json');
delete collection['linz:event_name'];
delete collection['linz:geographic_description];
const event = getEvent(collection);
const gd = getGeographicDescription(collection);
const target = generatePath('nz-imagery', 'urban-aerial-photos', gd, 'hawkes-bay', event, '2017-2018', '0.05m', '2193');
console.log(target);
// s3://nz-imagery/hawkes-bay/hawkes-bay_2017-2018_0.05m/rgb/2193/
```

Reply:
I've updated the tests to add this logic